snuba/admin/google.py

```diff
@@ -0,0 +1,49 @@
+from urllib.parse import urlencode
+
+import structlog
+from googleapiclient.discovery import Resource
+
+logger = structlog.get_logger().bind(module=__name__)
+
+
+def get_group_id(service: Resource, group_email: str) -> str:
+    try:
+        query_params = f"groupKey.id={group_email}"
+        request = service.groups().lookup()
+        request.uri += "&" + query_params
+        response = request.execute()
+        if "error" in response:
+            logger.exception(
+                f"An HTTP error occured when fetching group id for email {group_email}.",
+                google_api_error=response["error"],
+            )
+
+        return str(response.get("name", ""))
```

Comment:
```suggestion
        return response["name"]
```
When would `name` not be in the response? Should we log the error if that happens?

Reply:
This can happen when there's an error in the response in which case we log the error anyway. This ensures that we only log the error once in that case and not again when a KeyError is raised.